core/base/src/TApplication.cxx

```diff
@@ -468,7 +468,48 @@ void TApplication::GetOptions(Int_t *argc, char **argv)
          } else {
             Warning("GetOptions", "-e must be followed by an expression.");
          }
+      } else if (!strcmp(argv[i], "--")) {
+         TObjString* macro = nullptr;
+
+         if (fFiles) {
+            for (auto f = fFiles->GetEntriesFast(); f --> 0; ) {
```

Comment:
Ha! Funny syntax :-) Nonetheless - could you reword this to the more common (for ROOT)
```
for (auto f: *fFiles)
```